app/src/androidTest/java/tech/pauly/findapet/data/FakeAnimalRepository.java

```diff
@@ -0,0 +1,22 @@
+package tech.pauly.findapet.data;
+
+import android.support.test.espresso.idling.CountingIdlingResource;
+
+import io.reactivex.Observable;
+import tech.pauly.findapet.data.models.AnimalListResponse;
+
+public class FakeAnimalRepository extends AnimalRepository {
+
+    private CountingIdlingResource countingIdlingResource;
+
+    public FakeAnimalRepository(AnimalService animalService, ObservableHelper observableHelper, CountingIdlingResource countingIdlingResource) {
+        super(animalService, observableHelper);
+        this.countingIdlingResource = countingIdlingResource;
+    }
+
+    @Override
+    public Observable<AnimalListResponse> fetchAnimals() {
+        countingIdlingResource.increment();
```

Comment:
The key to the Espresso tests. This idling resource is global and a singleton so the test registers it and this class is substituted for the real AnimalRepository with the EspressoModule so the increment and decrement are executed to block the test thread.